packages/vscode-dependencies-validation/test/vscodeMocks.ts

```diff
@@ -0,0 +1,189 @@
+import { NPM_DEPENDENCY_ISSUES_CODE } from "../src/constants";
+import type {
+  Diagnostic,
+  DiagnosticCollection,
+  Uri,
+  CancellationToken,
+  CodeActionContext,
+  Event,
+  Position,
+  Range,
+  TextDocument,
+  TextLine,
+  CodeActionKind,
+} from "vscode";
+import { VscodeOutputChannel } from "../src/vscodeTypes";
+
+export const outputChannel: VscodeOutputChannel = {
+  append(value: string) {},
+  appendLine(value: string) {},
+  // @ts-expect-error -- https://stackoverflow.com/questions/68799234/typescript-pick-only-specific-method-from-overload-to-be-passed-to-parameterst
+  show(value?: boolean) {},
+};
+
+export const diagnosticCollection: DiagnosticCollection = {
+  name: "",
+  // @ts-expect-error -- https://stackoverflow.com/questions/68799234/typescript-pick-only-specific-method-from-overload-to-be-passed-to-parameterst
+  set: function (
+    uri: Uri,
+    diagnostics: readonly Diagnostic[] | undefined
+  ): void {
+    //throw new Error("Function not implemented.");
```

Comment:
if you can implement it naively with 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set

you can make clearer assertions during tests which would also be easier to debug (debug collection object vs debug mocks/spys)

Reply:
Not sure, let's talk about it